qiskit_nature/second_q/operators/polynomial_tensor.py

```diff
@@ -284,6 +283,7 @@ def todense(self) -> PolynomialTensor:
         if self.isdense():
             return self
 
+        _optionals.HAS_SPARSE.require_now("SparseArray")
```

Comment:
Could we even go as far as to change the `isinstance` check to `not isinstance(value, np.ndarray)` and in that case map it?
That would avoid relying on sparse in this matter altogether.

I guess the respective change could also be done in `tosparse` :thinking:

Reply:
The `tosparse` from the start already needs sparse just for having the `sparse_type` parameter which should be coming from the sparse library. Also, it calls `issparse` which needs `sparse`.
As for the `todense`, if you remove the explicit `SparseArray` test, you still need to call `todense` which is a method from sparse right ? without the `HAS_SPARSE` check it will raise a standard exception which we are trying to avoid by raising our own custom exception.